### ISSUE TITLE
get net::ERR_SPDY_PROTOCOL_ERROR

### ISSUE THREAD

**Issue by EnJiang** [timdp/es6-promise-pool#69]
I am developing an in-browser downloader.
Say I have 100 files on the server, I use promises and this pool to download each one of them and pack them into a zip file, and finally give it to users.
When the file is small, it works perfectly. However, when files are big, I got net::ERR_SPDY_PROTOCOL_ERROR from chrome.
If it helps, these are what's in the Network panel of chrome dev tool:
![image](https://user-images.githubusercontent.com/17062906/50471669-bb425b80-09f0-11e9-9d34-ed24354308e0.png)
![image](https://user-images.githubusercontent.com/17062906/50471683-cac1a480-09f0-11e9-8e80-2272d4d690b3.png)
... and after around 3 minutes of downloading and waiting, I got
![image](https://user-images.githubusercontent.com/17062906/50471774-1e33f280-09f1-11e9-84bd-c38c3e27d67f.png)
![image](https://user-images.githubusercontent.com/17062906/50471786-2be97800-09f1-11e9-99ce-2ababf1ad867.png)

**Comment by EnJiang** [comment]
Also, I am using Cloudflare if that matters.

**Comment by timdp** [comment]
It's unlikely that this is a bug in es6-promise-pool, but maybe I can help nonetheless.

I've seen `ERR_SPDY_PROTOCOL_ERROR` in Chrome under varying circumstances. I think it's a bit of a catch-all for unspecified SPDY errors and even HTTP/2 errors, unfortunately.

I also seem to recall that Chrome has a hardcoded limits of six sockets per destination IP. I'm seeing five concurrent requests in your screenshot so that could be related. (However, I thought SPDY allowed multiple requests per socket. I could be wrong though.)

What happens if you reduce the concurrency to five? If I'm understanding you correctly, you currently have it set to 100? That should be fine as long as the browser will allow you to do it, but maybe you're triggering some bug.

What happens if you try to run it in Firefox?

Finally, you should be able to verify that this also happens without es6-promise-pool by trying with e.g. [p-map](https://www.npmjs.com/package/p-map) instead. As I've mentioned under other issues, I've actually mostly stopped using this package myself since most scenarios can be achieved relatively easily nowadays using `async`/`await` and/or more lightweight libraries than mine. However, I'm happy to keep providing support for it.

**Comment by EnJiang** [comment]
Thank you for your response.
I tried various ways to solve this problem these days. I also noticed that it is not normal that all requests were made at the same time, but I checked and the concurrency limit was set to 5.
Finally, I stopped using Cloudflare and used Let's Encrypt instead, magically the problem was solved.
<br>
I have no idea what was happening behind this but I do hope my solution will shed some light for others facing this problem.
<br>
Since you as the author are encouraging users to turn to other lighweight libaries, I do think it is not proper to spend more time on such a strange case as it is.
Closing this issue now, and thank you for your work and kind support.
